### ISSUE TITLE
plugin will not recover after restored connection

### ISSUE THREAD

**Issue by chenbekor** [ReactiveCouchbase/ReactiveCouchbase-play#46]
how to reproduce:

start play app - fetch some data from a valid bucket - all is good
turn off networking - we now get TimeoutExceptions which is as expected
we turn networking back - reactivecouchbase will not recover

```java
play.api.UnexpectedException: Unexpected exception[ConfigurationException: Could not fetch a valid Bucket configuration.]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(ApplicationProvider.scala:166) ~[play_2.11-2.3.9.jar:2.3.9]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(ApplicationProvider.scala:130) ~[play_2.11-2.3.9.jar:2.3.9]
	at scala.Option.map(Option.scala:145) ~[scala-library-2.11.1.jar:na]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1.apply(ApplicationProvider.scala:130) ~[play_2.11-2.3.9.jar:2.3.9]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1.apply(ApplicationProvider.scala:128) ~[play_2.11-2.3.9.jar:2.3.9]
Caused by: com.couchbase.client.vbucket.ConfigurationException: Could not fetch a valid Bucket configuration.
	at com.couchbase.client.vbucket.provider.BucketConfigurationProvider.bootstrap(BucketConfigurationProvider.java:117) ~[couchbase-client-1.4.2.jar:1.4.2]
	at com.couchbase.client.vbucket.provider.BucketConfigurationProvider.getConfig(BucketConfigurationProvider.java:348) ~[couchbase-client-1.4.2.jar:1.4.2]
	at com.couchbase.client.CouchbaseConnectionFactory.getVBucketConfig(CouchbaseConnectionFactory.java:317) ~[couchbase-client-1.4.2.jar:1.4.2]
	at com.couchbase.client.CouchbaseClient.<init>(CouchbaseClient.java:257) ~[couchbase-client-1.4.2.jar:1.4.2]
	at org.reactivecouchbase.CouchbaseBucket.connect(Couchbase.scala:53) ~[reactivecouchbase-core_2.11-0.3-SNAPSHOT.jar:0.3-SNAPSHOT]
```

**Comment by chenbekor** [comment]
@mathieuancelin once you upgrade to 1.4.10 can you please check the above? it is urgent since the current status is we cannot recover from network partition issues.

**Comment by chenbekor** [comment]
fixed in 3e94759d66f32cb7c37f4422ac3f8b260641d2d4
